### DISCUSSION TITLE
Is it possible, without JDBC to link user table?

### DISCUSSION THREAD

**Discussion by kobukata** [keycloak/keycloak#24238] [Q&A] [answered]
I would like to link keycloak with the user table of an existing DB, but I would like to link it through an API without using JDBC. Is it possible?

**Comment by danielFesenmeyer** [comment, marked as answer]
Yes, it should be possible. As a starting point, you can have a look at this documentation: https://www.keycloak.org/docs/latest/server_development/index.html#_user-storage-spi

Note, there is a hint that the API is deprecated (legacy), but that's probably no longer true. (Because the mentioned map-storage will not replace the "legacy-store", at least not in the initially intended way - see https://www.keycloak.org/2023/10/map-store-removal)

**Comment by kobukata** [comment]
Thank you! I'll Try it ! :)
